databus-api/src/main/java/com/bazaarvoice/emodb/databus/api/DefaultSubscription.java

```diff
@@ -4,22 +4,51 @@
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonIgnore;
 import com.fasterxml.jackson.annotation.JsonProperty;
-import com.google.common.base.Objects;
 
 import java.time.Duration;
 import java.util.Date;
+import java.util.Objects;
+
+import static com.bazaarvoice.emodb.databus.api.Names.isLegalSubscriptionName;
 
 public final class DefaultSubscription implements Subscription {
+    private final static Duration EVENT_TTL_LIMIT = Duration.ofDays(365);
+
     private final String _name;
     private final Condition _tableFilter;
     private final Date _expiresAt;
     private final Duration _eventTtl;
 
     public DefaultSubscription(String name, Condition tableFilter, Date expiresAt, Duration eventTtl) {
-        _name = name;
+        _name = validateName(name);
         _tableFilter = tableFilter;
-        _expiresAt = expiresAt;
-        _eventTtl = eventTtl;
+        _expiresAt =  Objects.requireNonNull(expiresAt);
+        _eventTtl = validateEventTtl(eventTtl);
+    }
+
+    private static String validateName(final String name) {
+        Objects.requireNonNull(name);
```

Comment:
The NonNull checks are also placed in the caller when creating DefaultOwnedSubscription, but having them here too is a good idea.